apps/golang/install

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+
+[ -z "$GOROOT" ] && GOROOT="$HOME/.go"
+[ -z "$GOPATH" ] && GOPATH="$HOME/go"
+
+case "$arch" in
+  "64")
+    PLATFORM="linux-arm64"
+    ;;
+  "32")
+    PLATFORM="linux-armv6l"
+    ;;
+  *)
+    error "arch variable is not set, can not continue"
+    ;;
+esac
+
+if [ -z "$PLATFORM" ]; then
+    echo "Your operating system is not supported by the script."
+    exit 1
+fi
+
+if [ -n "$($SHELL -c 'echo $ZSH_VERSION')" ]; then
+    shell_profile="$HOME/.zshrc"
+elif [ -n "$($SHELL -c 'echo $BASH_VERSION')" ]; then
+    shell_profile="$HOME/.bashrc"
+elif [ -n "$($SHELL -c 'echo $FISH_VERSION')" ]; then
+    shell="fish"
+    if [ -d "$XDG_CONFIG_HOME" ]; then
+        shell_profile="$XDG_CONFIG_HOME/fish/config.fish"
+    else
+        shell_profile="$HOME/.config/fish/config.fish"
+    fi
+fi
+
+if [ ! -z "$1" ]; then
+    echo "Unrecognized option: $1"
+    exit 1
+fi
+
+if [ -d "$GOROOT" ]; then
+    echo "The Go install directory ($GOROOT) already exists. Exiting."
+    exit 1
+fi
+
+PACKAGE_NAME="go$version.$PLATFORM.tar.gz"
+TEMP_DIRECTORY=$(mktemp -d)
+
+echo "Downloading $PACKAGE_NAME ..."
+if hash wget 2>/dev/null; then
+    wget https://storage.googleapis.com/golang/$PACKAGE_NAME -O "$TEMP_DIRECTORY/go.tar.gz"
+else
+    curl -o "$TEMP_DIRECTORY/go.tar.gz" https://storage.googleapis.com/golang/$PACKAGE_NAME
+fi
+
+if [ $? -ne 0 ]; then
+    echo "Download failed! Exiting."
+    exit 1
+fi
+
+echo "Extracting File..."
+mkdir -p "$GOROOT"
+
+tar -C "$GOROOT" --strip-components=1 -xzf "$TEMP_DIRECTORY/go.tar.gz"
```

Comment:
you should also add error testing here (incase the extract command fails then pi-apps needs to report as an installation error)